packages/supabase_flutter/README.md

```diff
@@ -41,124 +41,213 @@ void main() async {
-    super.dispose();
-  }
+#### Native Apple Sign in
 
-  @override
-  Widget build(BuildContext context) {
-    return ElevatedButton(
-      onPressed: () {
-        supabase.auth.signInWithOtp(email: 'my_email@example.com');
-      },
-      child: const Text('Login'),
-    );
-  }
-}
+You need to [register your app ID with Apple](https://developer.apple.com/help/account/manage-identifiers/register-an-app-id/) with the `Sign In with Apple` capability selected, and add the bundle ID to your Supabase dashboard in `Authentication -> Providers -> Apple` before performing native Apple sign in.
+
+```dart
+// Perform Apple login on iOS and macOS
+await supabase.auth.signInWithApple();
 ```
 
-#### Native Sign in with Apple example
+`signInWithApple()` is only supported on iOS and on macOS. Use the `signInWithOAuth()` method to perform web-based Apple sign in on other platforms.
 
-Before you run the code, you need to [register your app ID with Apple](https://developer.apple.com/help/account/manage-identifiers/register-an-app-id/) with the `Sign In with Apple` capability selected, and add the bundle ID to your Supabase dashboard in `Authentication -> Providers -> Apple`. 
+#### Native Google sign in
 
-```dart
-import 'package:supabase_flutter/supabase_flutter.dart';
+You can perform native Google sign in on Android and iOS using [flutter_appauth](https://pub.dev/packages/flutter_appauth).
 
-final supabase = Supabase.instance.client;
+First, you need to create a client ID in your Google Cloud console and add them to your Supabase dashboard in `Authentication -> Providers -> Google -> Authorized Client IDs`. You can add multiple client IDs as comma separated string.
 
-return supabase.auth.signInWithApple();
-```
+- [Steps to obtain Android client ID](https://developers.google.com/identity/sign-in/android/start-integrating#configure_a_project)
+- [Steps to obtain iOS client ID](https://developers.google.com/identity/sign-in/ios/start-integrating#get_an_oauth_client_id)
 
-`signInWithApple()` is only supported on iOS and on macOS. Other platforms can use the `signInWithOAuth()` method to perform Apple login.
+Then add [flutter_appauth](https://pub.dev/packages/flutter_appauth) to your app. You also need [crypto](https://pub.dev/packages/crypto) package to hash nonce.
 
-The `signInWithApple` method is currently experimental and is subject to change. Follow [this issue](https://github.com/supabase/supabase-flutter/issues/399) for platform support progress.
+```bash
+flutter pub add flutter_appauth crypto
+```
 
+For iOS, no further setup is required. For Android, you also need to add the following in your `android/app/build.gradle` file. Your application ID cannot have underscores in it.
 
-### [Database](https://supabase.com/docs/guides/database)
+```
+...
+android {
+    ...
+    defaultConfig {
+        ...
+        manifestPlaceholders += [
+                'appAuthRedirectScheme': applicationId
+        ]
+    }
+}
+```
```

Comment:
@DanMossa So basically we probably should get rid of this section, right? Are there anything else in this PR that feel like is repeating the descriptions on the underlining dependencies?